### DISCUSSION TITLE
Docker performance optimization

### DISCUSSION THREAD

**Discussion by dantefromhell** [nextcloud/all-in-one#919] [Ideas] [answered]
I stumbled over this article on Reddit talking about quite impressive ~30% performance improvements by using sockets in between container instead of TCP/IP networking

https://old.reddit.com/r/selfhosted/comments/vf6jeg/i_used_unix_sockets_to_improve_the_performance_of/

Could this be an interesting improvement for this setup?

**Comment by szaimen** [comment, marked as answer]
I fear sockets are not really possible due to the architecture of this project or at least they would complicate the architecture by a lot.

**Reply by DaleJP** [reply]
I'd love some way of being able to tune it or have some inside tips on making it work better.
I'm having a horrible time with the front end and login being really slow and sometimes failing on laptops. I've tried to uninstall every unnecessary app, put this on a dedicated machine with a 4790k, 32G RAM, 10G NIC and RAID 0 2x1TB SSDs and my staff keep reporting that it won't log in so they can't work or it is too slow to work with. I use it every day on my PC and it's ok, but it seems whenever anyone else touches it, it messes up.
Is there a preferred browser? I use Chrome and they have Firefox. That is pretty much the only difference I see. It is driving me mad.

There is no doubt that this AIO is incredible for install, backup and ease of use but it is so slow in my particular case and I only have 4 regular users, 20 total. I am incredibly grateful for your work, I just need a road to increase performance.

The symptoms are generally failing to login with a spinning wheel and then slow navigation and a lot of lag within Collabora.

Any advice will really be helpful because it is becoming a general excuse for staff to put off work.

**Reply by szaimen** [reply]
Can you please open a new discussion for this? Thanks!

**Reply by dantefromhell** [reply]
True the sockets are bit trickier to setup than dockers `network` statements.
The additional benefit (besides performance) is that you make it impossible for IP/ firewall configuration to mess with the container communication.

Which might reduce user issues, especially given that the Linux kernel only has one routing & firewalling system that spans all interfaces - it's possible to create interference.
The sockets configuration is interference free in this regard
